### PR TITLE
functioning weatherAPI call

### PR DESCRIPTION
**Pull request by LukaTosic09**
*need to change name of OpenWeatherAPI.java to OpenMeteo.java

### CODE REVIEW

**Comment by LukaTosic09** [comment]
This was done locally on my computer, but Anna helped with it as well.

File: src/main/java/entities/ClothingItem.java [hunk 1]
```diff
@@ -9,6 +9,7 @@ public record ClothingItem(
         Image image,
         BodyPart bodyPart,
         Weather appropriateWeather,
-        Optional<String> description
+        Optional<String> description,
+        Boolean rainAppropriate
```
Comment (Artem-on-ishche) [inline src/main/java/entities/ClothingItem.java:13]: I think we should put the rain appropriateness inside the Weather class, but also rename it to something more meaningful, like WeatherAppropriateness. But either way, I think we should do this a bit later in another PR, as this one is only for implementing the weather API call

File: src/main/java/service/OpenWeatherAPI.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import java.io.IOException;
+
+public class OpenWeatherAPI implements WeatherDataSource {
+    @Override
+    public WeatherData getWeatherData() {
+        OkHttpClient client = new OkHttpClient().newBuilder().build();
+        Request request = new Request.Builder()
+                .url("https://api.open-meteo.com/v1/forecast?latitude=43.6532&longitude=-79.4163&" +
+                        "current=temperature_2m,precipitation&timezone=America%2FNew_York&forecast_days=1")
```
Comment (Artem-on-ishche) [inline src/main/java/service/OpenWeatherAPI.java:16]: You should change it to use the location data parameter after we merge my PR implementing the use case interactor

File: src/main/java/service/WeatherData.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
 package service;
 
-public record WeatherData(double temperature) {
+public record WeatherData(double temperature, boolean precipitation) {
```
Comment (Artem-on-ishche) [inline src/main/java/service/WeatherData.java:3]: I think you would also need to change the name of this variable for it to not cause merge conflicts

**Review by Artem-on-ishche — CHANGES_REQUESTED** [review]
Looks great! Now just change it to use location data so that there are no merge conflicts and then we can merge it into main

File: src/main/java/service/WeatherData.java (hunk 3 above)
Comment (LukaTosic09) [reply src/main/java/service/WeatherData.java:3]: yep

**Pull request closed without merging**